packages/file_selector/file_selector_ios/ios/Classes/FLTFileSelectorPlugin.m

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTFileSelectorPlugin.h"
+#import "FLTFileSelectorPlugin_Test.h"
+#import "messages.g.h"
+
+@implementation FLTFileSelectorPlugin
+
+#pragma mark - FLTFileSelectorApi
+
+- (void)openFileSelectorWithConfig:(FLTFileSelectorConfig *)config
+                        completion:(void (^)(NSArray<NSString *> * _Nullable,
+                                             FlutterError * _Nullable))completion {
+  if (self.pendingCompletion) {
+    completion(nil, [FlutterError errorWithCode:@"error"
+                                        message:@"There is already a pending file picker request."
+                                        details:nil]);
+    return;
+  }
+
+  UIDocumentPickerViewController *documentPicker = self.documentPickerViewControllerOverride ?:
+  [[UIDocumentPickerViewController alloc] initWithDocumentTypes:config.utis
+                                                         inMode:UIDocumentPickerModeImport];
+  documentPicker.delegate = self;
+  if (@available(iOS 11.0, *)) {
+    documentPicker.allowsMultipleSelection = config.allowMultiSelection.boolValue;
+  }
+
+  UIViewController *presentingVC = self.presentingViewControllerOverride ?:
+      UIApplication.sharedApplication.delegate.window.rootViewController;
+  if (presentingVC) {
+    [presentingVC presentViewController:documentPicker animated:YES completion:nil];
+    self.pendingCompletion = completion;
+  } else {
+    completion(nil, [FlutterError errorWithCode:@"error"
+                                        message:@"Missing root view controller."
+                                        details:nil]);
+  }
+}
+
+#pragma mark - FlutterPlugin
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  FLTFileSelectorPlugin *plugin = [[FLTFileSelectorPlugin alloc] init];
+  FLTFileSelectorApiSetup(registrar.messenger, plugin);
+}
+
+#pragma mark - UIDocumentPickerDelegate
+
+- (void)documentPicker:(UIDocumentPickerViewController *)controller
+  didPickDocumentAtURL:(NSURL *)url {
+  // This method is only called in iOS < 11.0.
+  if (self.pendingCompletion) {
+    self.pendingCompletion(@[ url.path ], nil);
+    self.pendingCompletion = nil;
```

Comment:
Rather than have three copies of this logic, please make a small helper (to reduce the chance of any future changes only doing the first step but not the second) that sends and then clears.

Reply:
Done. Using associated objects now.